File: JavaScript/intersection_of_two_linked_list.js
function ListNode(val, next) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? 0 : next)
}

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */

var getIntersectionNode = function (headA, headB) {
    l1 = 0
    l2 = 0
    list1 = headA
    list2 = headB

    while (list1 !== null) {
        l1++
        list1 = list1.next
    }

    while (list2 !== null) {
        l2++
        list2 = list2.next
    }
    list1 = headA
    list2 = headB

    if (l1 > l2) {
        dif = l1 - l2
        while (dif) {
            list1 = list1.next
            dif--
        }
    } else if (l1 < l2) {
        dif = l2 - l1
        while (dif) {
            list2 = list2.next
            dif--
        }
    }
    while (list1 !== list2) {
        list1 = list1.next
        list2 = list2.next
    }
    return list1
};

File: Go/intersection_of_two_linked_list.go
package main


type ListNode struct {
	Val  int
	Next *ListNode
}

func getIntersectionNode(headA, headB *ListNode) *ListNode {
	l1, l2 := 0, 0
	first, second := headA, headB

	for first != nil {
		l1++
		first = first.Next
	}
	for second != nil {
		l2++
		second = second.Next
	}
	first, second = headA, headB
	if l1 > l2 {
		dif := l1 - l2
		for dif > 0 {
			first = first.Next
			dif--
		}
	} else {
		dif := l2 - l1
		for dif > 0 {
			second = second.Next
			dif--
		}
	}
	for first != second {
		first = first.Next
		second = second.Next
	}
	return first
}